test/test_eos_vinet.cpp

```diff
@@ -0,0 +1,491 @@
+//------------------------------------------------------------------------------
+// © 2021-2022. Triad National Security, LLC. All rights reserved.  This
+// program was produced under U.S. Government contract 89233218CNA000001
+// for Los Alamos National Laboratory (LANL), which is operated by Triad
+// National Security, LLC for the U.S.  Department of Energy/National
+// Nuclear Security Administration. All rights in the program are
+// reserved by Triad National Security, LLC, and the U.S. Department of
+// Energy/National Nuclear Security Administration. The Government is
+// granted for itself and others acting on its behalf a nonexclusive,
+// paid-up, irrevocable worldwide license in this material to reproduce,
+// prepare derivative works, distribute copies to the public, perform
+// publicly and display publicly, and to permit others to do so.
+//------------------------------------------------------------------------------
+
+#include <array>
+#include <cmath>
+#include <cstdio>
+#include <cstdlib>
+#include <limits>
+#ifndef CATCH_CONFIG_RUNNER
+#include "catch2/catch.hpp"
+#endif
+
+#include <singularity-eos/base/constants.hpp>
+#include <singularity-eos/eos/eos.hpp>
+#include <test/eos_unit_test_helpers.hpp>
+
+using singularity::EOS;
+using singularity::Vinet;
+
+SCENARIO("Vinet EOS rho sie", "[VectorEOS][VinetEOS]") {
+  GIVEN("Parameters for a Vinet EOS") {
+    // Unit conversions
+    constexpr Real cm = 1.;
+    constexpr Real us = 1e-06;
+    constexpr Real Mbcc_per_g = 1e12;
+    // Vinet parameters for copper
+    constexpr Real rho0 = 8.93;
+    constexpr Real T0 = 298.0;
+    constexpr Real B0 = 1.3448466 * Mbcc_per_g;
+    constexpr Real BP0 = 4.956;
+    constexpr Real A0 = 5.19245e-05;
+    constexpr Real Cv0 = 0.383e-05 * Mbcc_per_g;
+    constexpr Real E0 = 0.0;
+    constexpr Real S0 = 5.05e-04 * Mbcc_per_g;
+    constexpr Real d2to40[39] = {0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0.,
+                                 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0.,
+                                 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0.};
+    // Create the EOS
+    EOS host_eos = Vinet(rho0, T0, B0, BP0, A0, Cv0, E0, S0, d2to40);
+    EOS eos = host_eos.GetOnDevice();
```

Comment:
@aematts the type of `host_eos` and `eos` should be `Vinet` for the time being until we can get the `Variant` class to have all of the functionality we need